assembly/pom.xml

```diff
@@ -81,8 +60,75 @@
           <skip>true</skip>
         </configuration>
       </plugin>
-
       </plugins>
   </build>
+  <profiles>
+    <profile>
+      <id>targz</id>
+      <activation><property><name>targz</name></property></activation>
+      <build>
+        <plugins>
+
+          <plugin>
+            <artifactId>maven-assembly-plugin</artifactId>
+            <configuration>
+              <appendAssemblyId>false</appendAssemblyId>
+              <descriptors>
+                <descriptor>assembly.xml</descriptor>
+              </descriptors>
+            </configuration>
+            <executions>
+              <execution>
+                <id>livy-assembly</id>
+                <phase>package</phase>
+                <goals>
+                  <goal>single</goal>
+                </goals>
+                <configuration>
+                  <finalName>${assembly.name}</finalName>
+                  <formats>
+                    <format>tar.gz</format>
```

Comment:
Instead of all this copy & paste, you could have `tar.gz` be a variable (e.g. `${assembly.format}`) and let the profile override the property (or even leave it for the user to do that on the command line).

That's a much, much smaller change.